src/command_line/indexing/index_sdf.rs

```diff
@@ -78,75 +79,157 @@ pub fn action(matches: &ArgMatches) -> eyre::Result<usize> {
 
-        let (canon_taut, fp, computed) = process_result.unwrap();
-
-        let json: serde_json::Value = serde_json::to_value(&computed)?;
-        let descriptors_map: Map<String, Value> = if let serde_json::Value::Object(map) = json {
-            map
-        } else {
-            panic!("not an object");
-        };
-
-        let mut doc = doc!(
-            smiles_field => canon_taut.as_smiles(),
-            fingerprint_field => fp.0.into_vec()
-        );
-
-        for field in KNOWN_DESCRIPTORS {
-            if let Some(serde_json::Value::Number(val)) = descriptors_map.get(field) {
-                let current_field = *descriptors_fields
-                    .get(field)
-                    .ok_or(eyre::eyre!("Failed to extract field"))?;
-                let current_value = val
-                    .as_f64()
-                    .ok_or(eyre::eyre!("Failed to convert descriptor to float"))?;
-                if field.starts_with("Num") || field.starts_with("lipinski") {
-                    doc.add_field_value(current_field, current_value as i64);
-                } else {
-                    doc.add_field_value(current_field, current_value);
-                };
+            if counter > 0 && counter % 10_000 == 0 {
+                println!("{:?} compounds processed so far", counter);
             }
         }
+    }
 
-        let scaffold_matches = scaffold_search(&canon_taut, &PARSED_SCAFFOLDS)?;
+    if !mol_vec.is_empty() {
+        let _ = mol_vec
+            .clone()
+            .into_par_iter()
+            .map(|m| {
+                let doc = create_tantivy_doc(
+                    m,
+                    smiles_field,
+                    fingerprint_field,
+                    &descriptor_fields,
+                    extra_data_field,
+                );
+
+                match doc {
+                    Ok(doc) => {
+                        let write_operation = index_writer.add_document(doc);
+
+                        match write_operation {
+                            Ok(_) => (),
+                            Err(_) => {
+                                println!("Failed doc creation");
+                                let mut num = failed_counter.lock().unwrap();
+                                *num += 1;
+                            }
+                        }
+                    }
+                    Err(_) => {
+                        println!("Failed doc creation");
+                        let mut num = failed_counter.lock().unwrap();
+                        *num += 1;
+                    }
+                }
+            })
+            .collect::<Vec<_>>();
+
+        index_writer.commit()?;
+        counter += mol_vec.len();
+    }
 
-        let scaffold_json = match scaffold_matches.is_empty() {
-            true => serde_json::json!({"scaffolds": vec![-1]}),
-            false => serde_json::json!({"scaffolds": scaffold_matches}),
-        };
+    println!(
```

Comment:
You should probably replace all instances of `println!()` with `log::info!()` or `log::warning!()`, etc. That way we can have different levels of output which can be filtered using `RUST_LOG` env variables.